app/src/main/java/com/example/bookclub/QRCodeMainActivity.java

```diff
@@ -84,6 +103,24 @@ private void bindCameraPreview(@NonNull ProcessCameraProvider cameraProvider) {
 
         preview.setSurfaceProvider(mPreviewView.createSurfaceProvider());
 
-        Camera camera = cameraProvider.bindToLifecycle((LifecycleOwner) this, cameraSelector, preview);
+        ImageAnalysis imageAnalysis =
+                new ImageAnalysis.Builder()
+                        .setTargetResolution(new Size(1280, 720))
+                        .setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
+                        .build();
+
+        imageAnalysis.setAnalyzer(ContextCompat.getMainExecutor(this), new QRCodeImageAnalyzer(new QRCodeFoundListener() {
+            @Override
+            public void onQRCodeFound(String _qrCode) {
+                qrCode = _qrCode;
+                mQRCodeFound.setVisibility(View.VISIBLE);
+            }
+
+            @Override
+            public void qrCodeNotFound() {
+                mQRCodeFound.setVisibility(View.INVISIBLE);
```

Comment:
GONE instead of invisible. 
GONE is better because the invisible views are still measured.